src/pqm/pqm.py

```diff
@@ -312,26 +687,43 @@ def pqm_chi2(
     float or list
         chi2 statistic(s).
     """
+
+    # Check the device and convert to the respective type (Numpy or Torch) and call their respective _pqm_test function
+    if device.type == 'cpu':
```

Comment:
same here

Reply:
Good catch. I have updated it to be if device.type == 'cpu' or device == 'cpu':